client/routes/workflow/components/cytoscape-graph.vue

```diff
@@ -0,0 +1,237 @@
+<template>
+  <div id="cytoscape">
+    <div ref="cy" id="cy"></div>
+  </div>
+</template>
+
+<script>
+import dagre from "cytoscape-dagre";
+import { getEventConnections } from "../helpers/get-event-connections";
+import graphStyles from "../helpers/graph-styles";
+import store from "../../../store/index";
+import cytoscape from "cytoscape";
+import omit from "lodash-es/omit";
+
+cytoscape.use(dagre);
+
+export default {
+  name: "cytoscape-graph",
+  props: ["events"],
+  data() {
+    return {
+      nodes: [],
+      edges: [],
+      styles: graphStyles,
+      parentArray: []
+    };
+  },
+  watch: {
+    selectedEvent(id) {
+      if (id) this.selectNode(id);
+    }
+  },
+  methods: {
+    zoomToNode(node) {
+      let zoom = 1.1,
+        bb = node.boundingBox(),
+        w = cy.width(),
+        h = cy.height(),
+        pan = {
+          x: (w - zoom * (bb.x1 + bb.x2)) / 2,
+          y: (h - zoom * (bb.y1 + bb.y2)) / 2
+        };
+
+      //Pan the graph to view node
+      cy.animate({
+        zoom: 1.1,
+        pan: pan
+      });
+    },
+    updateChildBtn(node) {
+      let nodeData = node.data();
+      if (nodeData.childRoute) {
+        store.commit("childRoute", {
+          route: nodeData.childRoute,
+          btnText: "To child"
+        });
+      } else if (nodeData.newExecutionRunId) {
+        store.commit("childRoute", {
+          route: nodeData.newExecutionRunId,
+          btnText: "Next execution"
+        });
+      } else {
+        store.commit("toggleChildBtn");
+      }
+    },
+    selectNode(id) {
+      //Deselect all previously selected nodes
+      cy.$(":selected").deselect();
+
+      //Mark current node as selected
+      let node = cy.elements("node#" + id);
+      node.select();
+
+      this.updateChildBtn(node);
```

Comment:
I would perhaps reshape what this function does.
On select of a node you could store the node in your store, then in your render function react off that if that makes sense. 
So try to move your logic within a component's event to its render function based on data set in your store.
That way if other things need to react off a node selection, it can as the node information is already set in the store (for example populating node events panel).

Reply:
Not sure what you mean with moving from event to its render function here.

Should I add a watcher if that value is updated in the store you mean?